### DISCUSSION TITLE
How to monitor is device charging and battery level?

### DISCUSSION THREAD

**Discussion by dhoineProg** [AvaloniaUI/Avalonia#16625] [Q&A] [answered]
I have a project where I need to monitor the battery level of an device and whether the device is charging. How can I do this with avalonia and is it even possible to do this with avalonia?

**Comment by thevortexcloud** [comment, marked as answer]
If you use Xamarin/MAUI Essentials you can.

https://learn.microsoft.com/en-us/previous-versions/dotnet/api/xamarin.essentials.battery?view=xamarin-essentials-android
https://learn.microsoft.com/en-us/dotnet/maui/platform-integration/device/battery?view=net-maui-8.0&tabs=android

I don't think it works on Linux though? But reading the battery info there is as simple as reading from `/sys/class/power_supply/BAT0/capacity` and friends.

**Reply by dhoineProg** [reply]
Thank you so much for your answer! It was extremely helpful in solving my problem.

I used Xamarin.Essentials, and this allowed my project to start working.
